src/state/selectors/buildInferredResults.ts

```diff
@@ -16,11 +16,24 @@ export default function buildInferredResults<
     if (id === undefined || id === '') return null;
     return id as any;
   }
-  if (schema instanceof schemas.Array || Array.isArray(schema)) {
+  if (
+    schema instanceof schemas.Array ||
+    Array.isArray(schema) ||
+    schema instanceof schemas.Values
+  ) {
     // array schemas should not be inferred because they're likely to be missing many members
+    // Values cannot be inferred because they have aribtrary keys
     return null;
   }
-  // TODO: handle other schemas besides array and object
+  if (schema instanceof schemas.Union) {
+    const discriminatedSchema = schema.inferSchema(params, undefined, '');
```

Comment:
this requires sending the type discriminator which may not always be natural, but can be an easy addition to optimize specific scenarios